### PR TITLE
change the condition of waiting for the OSD pod in the test "test_automated_recovery_from_stopped_node_and_start"

### LINKED ISSUE #4766: In the test "test_automated_recovery_from_stopped_node_and_start" change the condition of waiting for the OSD pod
In the test "test_automated_recovery_from_stopped_node_and_start", in line https://github.com/red-hat-storage/ocs-ci/blob/master/tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_IPI.py#L264 `wait_for_resource_state(temp_osd, constants.STATUS_TERMINATING, timeout=240)` we wait for the OSD pod to reach the Terminating state. In 2 tests I ran recently, this condition has failed. In this run, https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/5415/, the test https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/5415/testReport/tests.manage.z_cluster.nodes.test_automated_recovery_from_failed_nodes_reactive_IPI/TestAutomatedRecoveryFromStoppedNodes/test_automated_recovery_from_stopped_node_and_start_False_/ failed because the OSD pod did not reach the "Terminating" condition. But when I looked at the rook-ceph-operator logs http://magna002.ceph.redhat.com/ocsci-jenkins/openshift-clusters/ikave-aws48-test/ikave-aws48-test_20210823T081224/logs/failed_testcase_ocs_logs_1629714519/test_automated_recovery_from_stopped_node_and_start%5bFalse%5d_ocs_logs/ocs_must_gather/quay-io-rhceph-dev-ocs-must-gather-sha256-2d09a8df6c012a4a6383c32f5ba190fb4e00a75fd90a99be3378ab051e583b07/namespaces/openshift-storage/pods/rook-ceph-operator-6477dddcdf-4g4zb/rook-ceph-operator/rook-ceph-operator/logs/current.log I saw that it reached the status Terminating: 

> 2021-08-23T13:28:56.289488722Z 2021-08-23 13:28:56.289460 I | op-k8sutil: force deleting pod "rook-ceph-osd-0-78d45bdfd8-96cfg" that appears to be stuck terminating
2021-08-23T13:28:56.322064553Z 2021-08-23 13:28:56.322042 I | op-k8sutil: pod "rook-ceph-osd-0-78d45bdfd8-96cfg" deletion succeeded

And in the console output, the error was just 30 seconds before:

> 13:28:24 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage get Pod rook-ceph-osd-0-78d45bdfd8-96cfg -n openshift-storage
13:28:29 - MainThread - ocs_ci.ocs.ocp - INFO - status of rook-ceph-osd-0-78d45bdfd8-96cfg at column STATUS was Running, but we were waiting for Terminating
13:28:29 - MainThread - ocs_ci.ocs.ocp - ERROR - timeout expired: Timed out after 240s running get("rook-ceph-osd-0-78d45bdfd8-96cfg", True, None)

Same thing I saw in the second run of the test. 

In the second run, https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/5466/, the test https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/5466/testReport/tests.manage.z_cluster.nodes.test_automated_recovery_from_failed_nodes_reactive_IPI/TestAutomatedRecoveryFromStoppedNodes/test_automated_recovery_from_stopped_node_and_start_True_/ failed because it didn't find the OSD pod. When looking at the rook-ceph-operator logs http://magna002.ceph.redhat.com/ocsci-jenkins/openshift-clusters/ikave-aws48-test/ikave-aws48-test_20210824T165900/logs/failed_testcase_ocs_logs_1629831382/test_automated_recovery_from_stopped_node_and_start%5bTrue%5d_ocs_logs/ocs_must_gather/quay-io-rhceph-dev-ocs-must-gather-sha256-2d09a8df6c012a4a6383c32f5ba190fb4e00a75fd90a99be3378ab051e583b07/namespaces/openshift-storage/pods/rook-ceph-operator-6477dddcdf-4xc6w/rook-ceph-operator/rook-ceph-operator/logs/current.log, we can see that the pod was stuck in the Terminating state and then was deleted: 

> 2021-08-24T21:40:44.732255203Z 2021-08-24 21:40:44.732231 I | op-k8sutil: force deleting pod "rook-ceph-osd-0-5b55547c98-p5586" that appears to be stuck terminating
2021-08-24T21:40:44.759663811Z W0824 21:40:44.759640       7 warnings.go:70] batch/v1beta1 CronJob is deprecated in v1.21+, unavailable in v1.25+; use batch/v1 CronJob
2021-08-24T21:40:44.832328742Z 2021-08-24 21:40:44.832303 I | op-k8sutil: pod "rook-ceph-osd-0-5b55547c98-p5586" deletion succeeded

It seems that the diff between the OSD pod reached the Terminating state to the deletion of the OSD pod was no more than 20 seconds, and that's why in the test, it failed to "catch" the OSD pod in the Terminating state.

### PR DESCRIPTION
**Pull request by yitzhak12**
Fixes #4766

### CODE REVIEW

**Comment by ebenahar** [comment]
@yitzhak12 please add either description or an issue which this PR fixes

**Comment by yitzhak12** [comment]
fix #4766

**Review by ebenahar — APPROVED** [review]

File: ocs_ci/helpers/helpers.py [hunk 1]
```diff
@@ -3235,12 +3235,9 @@ def check_osd_log_exist_on_rook_ceph_operator_pod(
...
     rook_ceph_operator_logs = get_logs_rook_ceph_operator()
     for line in rook_ceph_operator_logs.splitlines():
-        if re.search(r"\d{4}-\d{2}-\d{2}", line):
-            log_date_time_obj = datetime.datetime.strptime(
-                line[:26], "%Y-%m-%d %H:%M:%S.%f"
-            )
-            if log_date_time_obj > last_log_date_time_obj:
-                new_logs.append(line)
+        log_date_time_obj = get_event_line_datetime(line)
+        if log_date_time_obj and log_date_time_obj > last_log_date_time_obj:
```
Comment (PrasadDesala) [inline ocs_ci/helpers/helpers.py:3239]: why are we calling same object twice here with an and operator?

**Review by PrasadDesala — APPROVED** [review]

**Review by AviadP — APPROVED** [review]

**Review by annagitel — APPROVED** [review]
LGTM

**Pull request merged**